omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java

```diff
@@ -110,11 +114,24 @@ public String get(PageModel model,
 			Context.removeProxyPrivilege(GET_LOCATIONS);
 		}
 
+        model.addAttribute("showSessionLocations", !isLocationUserPropertyAvailable(administrationService));
+		Object showLocation = pageRequest.getAttribute("showSessionLocations");
+		if(showLocation != null && showLocation.toString().equals("true")) {
+            model.addAttribute("showSessionLocations", true);
+        }
 		model.addAttribute("lastSessionLocation", lastSessionLocation);
 
 		return null;
 	}
 
+	private boolean isLocationUserPropertyAvailable(AdministrationService administrationService) {
+		String showLocationSelector = administrationService.getGlobalProperty(ReferenceApplicationConstants.LOCATION_USER_PROPERTY_NAME);
```

Comment:
What if it is available but does not have a value of true?

Reply:
Actually, I decided to use only one GP for this task, So if that GP contains any value except "false", it will return that value as the userProperty name

Reply:
You have a gp named "referenceapplication.locationUserPropertyName" and then expect a value of true or false? This does not look intuitive because from the look of the gp name, i would expect it to contain the actual name.

Reply:
We decided to use one GP for this, So when evet it is set, we know that it true, if not set or has value as "false" we know it is false.

Reply:
What happens when it has a value of a non existing property?